pyvista/core/filters/data_set.py

```diff
@@ -7057,18 +7057,15 @@ def transform(
             self.cell_data.active_scalars_name = active_cell_scalars_name
             res.cell_data.active_scalars_name = active_cell_scalars_name
 
-        if inplace:
-            self.copy_from(res, deep=False)
-            return self
-
+        self_output = self if inplace else self.__class__()
+        output = pyvista.StructuredGrid() if isinstance(self, pyvista.Grid) else self_output
         # The output from the transform filter contains a shallow copy
         # of the original dataset except for the point arrays.  Here
         # we perform a copy so the two are completely unlinked.
-        if isinstance(self, pyvista.Grid):
-            output: _vtk.vtkDataSet = pyvista.StructuredGrid()
+        if inplace:
+            output.copy_from(res, deep=False)
         else:
-            output = self.__class__()
-        output.copy_from(res, deep=True)
+            output.copy_from(res, deep=True)
```

Comment:
Just to confirm, are the behavior will change if `inplace` when self is `pyvista.Grid`?

Reply:
No the logic should be the same: shallow copy if inplace, deep copy otherwise. And output SteucturedGrid if input is a Grid (ImageData or RectilinearGrid)

Reply:
inplace must always be false for Grid inputs, there is logic on the first line of this method to raise an error for these types if inplace is True

Reply:
Ah. That makes sense. Thanks!